src/net_processing.cpp

```diff
@@ -1054,8 +1051,8 @@ bool PeerManager::MaybePunishNodeForBlock(NodeId nodeid, const BlockValidationSt
             }
 
             // Discourage outbound (but not inbound) peers if on an invalid chain.
-            // Exempt HB compact block peers and manual connections.
-            if (!via_compact_block && !node_state->m_is_inbound && !node_state->m_is_manual_connection) {
+            // Exempt HB compact block peers. Manual connections are always protected from discouragement.
```

Comment:
I don't think the comment "Manual connections are always protected from discouragement." should be here since it's not specific to this site.

Reply:
Bookmarked, I'll take it if further changes are required.